core/src/test/java/com/adobe/aem/guides/wknd/core/models/impl/PageSearchImplTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Created on 24-08-2023 22:39 by jjedlins
+ *
+ * Copyright (c) 2001-2023 Unity S.A.
+ * ul. Strzegomska 2-4, 53-611 Wrocław, Poland
+ * Wszelkie prawa zastrzeżone
+ *
+ * Niniejsze oprogramowanie jest własnością Unity S.A.
+ * Wykorzystanie niniejszego oprogramowania jest możliwe tylko na podstawie
+ * i w zgodzie z warunkami umowy licencyjnej zawartej z Unity S.A.
+ */
+
+package com.adobe.aem.guides.wknd.core.models.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.adobe.aem.guides.wknd.core.models.PageSearch;
+
+import io.wcm.testing.mock.aem.junit5.AemContext;
+import io.wcm.testing.mock.aem.junit5.AemContextExtension;
+
+
+@ExtendWith({AemContextExtension.class, MockitoExtension.class })
+class PageSearchImplTest {
```

Comment:
I think it should be useful for you to learn how to set up a service in mocked aem context. 
I suggest you to try to 
1. mock the return value for PageSearchService.executeQuery(...)
2. set up mocked PageSearchService in aemcontext
3. test if it works for PageSearch model

Reply:
try to find some info's / examples on our own, in case u feel stuck pls contact me directly